### PR TITLE
feat: Support if- and while- let chains

### LINKED ISSUE #11320: Add support for let chains
See https://github.com/rust-lang/rust/pull/88642.

### PR DESCRIPTION
**Pull request by ChayimFriedman2**
Closes #11320.

### CODE REVIEW

**Comment by ChayimFriedman2** [comment]
@Veykril What with this?

**Comment by Veykril** [comment]
Will take a look at this once I get time as this is a larger PR to review

**Comment by ChayimFriedman2** [comment]
Fine 😃 I just wanted to make sure this wasn't forgotten.

File: crates/syntax/src/ast/node_ext.rs [hunk 1]
```diff
@@ -533,9 +533,39 @@ impl ast::Item {
...
+                        .or_else(|| expr.rhs().map(contains_let))
+                        .unwrap_or(false)
+                }
+                ast::Expr::ParenExpr(expr) => expr.expr().map_or(false, contains_let),
+                ast::Expr::LetExpr(_) => true,
+                _ => false,
+            }
+        }
     }
 }
```
Comment (Veykril) [inline crates/syntax/src/ast/node_ext.rs:570]: Let's move these into `ide_db::helpers::node_ext` as free functions, as the methods in this file here `should only do simple, shallow tasks related to the syntax of the node itself.` as per the comment at the top

**Review by Veykril — COMMENTED** [review]
Lgtm with one small nit

**Comment by Veykril** [comment]
Thanks!
bors r+

**Pull request merged**